src/executor/index_scan_executor.cpp

```diff
@@ -195,92 +208,66 @@ bool IndexScanExecutor::ExecPrimaryIndexLookup() {
             transaction_manager.SetTransactionResult(RESULT_FAILURE);
             return res;
           }
+          
         } else {
           expression::ContainerTuple<storage::TileGroup> tuple(
-              tile_group.get(), tuple_location.offset);
+            tile_group.get(), tuple_location.offset);
           auto eval =
-              predicate_->Evaluate(&tuple, nullptr, executor_context_).IsTrue();
+            predicate_->Evaluate(&tuple, nullptr, executor_context_).IsTrue();
           if (eval == true) {
-            visible_tuples[tuple_location.block]
-                .push_back(tuple_location.offset);
+            visible_tuples[tuple_location.block].push_back(tuple_location.offset);
 
             auto res = transaction_manager.PerformRead(tuple_location);
             if (!res) {
               transaction_manager.SetTransactionResult(RESULT_FAILURE);
               return res;
             }
+            
           }
         }
         break;
       }
-      // if the tuple is not visible.
+        // if the tuple is not visible.
       else {
-        ItemPointer old_item = tuple_location;
-        cid_t old_end_cid = tile_group_header->GetEndCommitId(old_item.offset);
 
+        if (tile_group_header->GetTransactionId(tuple_location.offset) == INITIAL_TXN_ID
+          && tile_group_header->GetEndCommitId(tuple_location.offset) <= concurrency::current_txn->GetBeginCommitId()) {
```

Comment:
Don't write huge predicate checks within the if condition. Split it into multiple local variables.